### DISCUSSION TITLE
[v4] Animation with ::before Not Working

### DISCUSSION THREAD

**Discussion by robertcorponoi** [tailwindlabs/tailwindcss#15337] [Help] [answered]
I have an animation that starts as a circle at the top of the screen and then fills up the page with a background.

This animation works in v3 but in v4, nothing happens. Inspecting the class names on the `div` and its `::before` pseduo-element, it looks like it matches the same ones that are present in the v3 version. 

I've looked at:
- [Stacking order-sensitive variants](https://tailwindcss.com/docs/v4-beta#stacking-order-sensitive-variants)
- [[v4] Styling doesn't get applied properly to ::after elements](https://github.com/tailwindlabs/tailwindcss/issues/15241)

But I don't think they apply to this case.

I created a [tailwindPlay](https://play.tailwindcss.com/RmQkuLs50i) for it. You can switch between v3 and v4 versions and see it works in v3 but not in v4.

**Comment by wongjn** [comment, marked as answer]
You'd want to override the `before:translate-y-[9999px]` class in the `@keyframes` using the `translate` property instead of the `transform` property: https://play.tailwindcss.com/pjPrQz9jqb?file=css

**Reply by robertcorponoi** [reply]
You're right, I completely missed that. Thank you!
